flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/generated/CompileUtils.java

```diff
@@ -62,7 +62,7 @@
         try {
             Cache<ClassLoader, Class> compiledClasses =
                     COMPILED_CACHE.get(
-                            name,
+                            name + code,
```

Comment:
If we want to take code into account, we only need to use code as the key, because code already include the class name. This can avoid creating a new String as the code may be very big. 

Would be better to add a comment above this to explain why we use code instead of name as the cache key.

Reply:
I think the runtime overhead is neglectable because we are not in the hotpath but pre-flight phase. But sure, code as a key should be sufficient. I will adapt the PR and add a comment.